src/sp/it/pl/main/AppUtil.kt

```diff
@@ -0,0 +1,29 @@
+@file:JvmName("AppUtil")
+
+package sp.it.pl.main
+
+import javafx.application.Application
+import sp.it.pl.util.file.Util.isValidatedDirectory
+import sp.it.pl.util.file.childOf
+import java.io.File
+import java.net.URLConnection
+
+lateinit var APP: App
+
+val isAPPInitialized
```

Comment:
My initial reaction was that this belongs to the App.class, but I see this is actually invoking the isInitialized on the very Kotlin property (APP). I very much do not like this approach here because it is very arbitrary (not only that is is also confusing because the application is still not necessarily initialized if it returns true, only the field is). Are we going to provide more fields in the future when we need to access APP instance before initialization so we can abuse them to provide 'sensible' defaults? No.

I think the good solution is to initialize the APP field as the very first thing after the App is instantiated (and this is important: before any of its fields are initialized). This can be done by adding a synthetic field as a first field in the App.class and have this field's initialization to set AppUtils.APP instead of doing this in the constructor, which is too late.

I.e. solution I suggested in #87

Btw, either way, we are still at the mercy of initialization order, we always will be unless we express all dependencies through the constructor (which we will not). The issue is that even if APP field is initialized, if there is wrong component initialization order or cyclic dependency, we are back to exceptions. We will have to live with this.

Reply:
Let me add that the dependency problem could be solved by using a dependency injection framework. The benefit is that the framework manages the order of component initialization. The trade off is added boilerplate as well as hidden magic.